pkg/workflow_execution.go

```diff
@@ -1463,3 +1463,28 @@ func (c *Client) getWorkflowExecutionAndTemplate(namespace string, uid string) (
 
 	return
 }
+
+func (c *Client) DeleteWorkflowExecution(namespace, uid string) error {
+	err := c.ArgoprojV1alpha1().Workflows(namespace).Delete(uid, nil)
+	if err != nil {
+		if strings.Contains(err.Error(), "not found") {
+			return nil
+		}
+		return err
+	}
+	return nil
+}
+
+func (c *Client) DeleteWorkflowExecutionDb(workflowUid string) error {
+	query, args, err := sb.Delete("workflow_executions").Where(sq.Eq{
+		"uid": workflowUid,
+	}).ToSql()
+	if err != nil {
+		return err
+	}
+
+	if _, err := c.DB.Exec(query, args); err != nil {
```

Comment:
This should be
```go
c.DB.Exec(query, args...)
```

It trips me up all the time >.<

Reply:
This should be:

```
err := sb.Delete("workflow_executions").Where(sq.Eq{
	"uid": workflowUid,
         "namespace": namespace,
  }).RunWith(c.DB).Exec()
```
Since we are not marshalling into a struct. Also note the namespace in `Where` so we need to pass that here.

Another note, I don't see a point to having a separate DB method here. This is a small and query can easily be placed in the calling method. I don't see a case where we will delete only database records and not the corresponding K8S resource.

Reply:
Refactored per feedback.